From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rong Bao <rong.bao@csmantle.top>
Date: Sun, 30 Aug 2026 18:36:11 +0800
Subject: [PATCH 1/2] BACKPORT: meson.build: use operator "/" for path
 concatenation

[ Upstream commit a20f0a57f228a9e1f1a9cd0ce1f099a09bb0131d ]
Link: https://github.com/mchehab/rasdaemon/commit/a20f0a57f228a9e1f1a9cd0ce1f099a09bb0131d
Link: https://github.com/mchehab/rasdaemon/pull/263 ("meson.build: fix
 prefixed path handling to match Meson expectations - #263")
Signed-off-by: Rong Bao <rong.bao@csmantle.top>
---
 meson.build | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/meson.build b/meson.build
index bdcaad1d83d56199fbbabce0e61af32b85bb7a84..2a7dd9db488f45aec87919bfec9d30713d3ecb80 100644
--- a/meson.build
+++ b/meson.build
@@ -578,17 +578,17 @@ ras_mc_ctl = configure_file(
 
 rasdaemon_env = prefix / get_option('sysconfdir') / 'sysconfig' / 'rasdaemon'
 install_data('misc/rasdaemon.env',
-    install_dir: get_option('sysconfdir') + '/sysconfig',
+    install_dir: get_option('sysconfdir') / 'sysconfig',
     rename: 'rasdaemon',
 )
 
 install_data('contrib/mc_event_trigger',
-    install_dir: get_option('sysconfdir') + '/ras/triggers',
+    install_dir: get_option('sysconfdir') / 'ras/triggers',
     install_mode: 'rwxr-xr-x',
 )
 
 install_data('contrib/mem_fail_trigger',
-    install_dir: get_option('sysconfdir') + '/ras/triggers',
+    install_dir: get_option('sysconfdir') / 'ras/triggers',
     install_mode: 'rwxr-xr-x',
 )
 
@@ -605,7 +605,7 @@ conf_data = {
     'sysconfdir' : get_option('sysconfdir'),
     'localstatedir' : get_option('localstatedir'),
     'sbindir' : get_option('sbindir'),
-    'SYSCONFDEFDIR' : get_option('sysconfdir') + '/sysconfig',
+    'SYSCONFDEFDIR' : get_option('sysconfdir') / 'sysconfig',
 
     'META_DATE' : run_command('date', '+%d %B %Y', check: true).stdout().strip(),
     'META_ALIAS' : meson.project_name(),
@@ -635,7 +635,7 @@ install_map = {
     'misc/rasdaemon.spec.in' : '',
 
     # Man pages
-    'man/rasdaemon.8.in' : get_option('mandir') + '/man8',
+    'man/rasdaemon.8.in' : get_option('mandir') / 'man8',
 
     # Systemd service files
     'misc/rasdaemon.service.in' : systemd_unitdir,
@@ -645,9 +645,9 @@ install_map = {
     'util/ras_config.py.in': pydir,
 
     # Other files
-    'misc/rasdaemon.syslog-ng.in' : get_option('sysconfdir') + '/syslog-ng/conf.d',
-    'misc/rasdaemon.rsyslog.in' : get_option('sysconfdir') + '/rsyslog.d',
-    'misc/rasdaemon.logrotate.in' : get_option('sysconfdir') + '/logrotate.d',
+    'misc/rasdaemon.syslog-ng.in' : get_option('sysconfdir') / 'syslog-ng/conf.d',
+    'misc/rasdaemon.rsyslog.in' : get_option('sysconfdir') / 'rsyslog.d',
+    'misc/rasdaemon.logrotate.in' : get_option('sysconfdir') / 'logrotate.d',
 }
 
 # File renaming install rules
@@ -719,7 +719,7 @@ endforeach
 
 if labels_list.length() > 0
     install_data(labels_list,
-        install_dir: get_option('sysconfdir') + '/ras/dimm_labels.d',
+        install_dir: get_option('sysconfdir') / 'ras/dimm_labels.d',
         install_mode: 'rw-r--r--',
     )
 endif

base-commit: 06e30a7ca3b2f66d98446c93950695eb1d652b40
-- 
2.55.0

